nixos/lib/make-disk-image.nix

```diff
@@ -266,6 +263,23 @@ let format' = format; in let
           $diskImage
       ''}
     '';
+    efixbootldr = ''
+      parted --script $diskImage -- \
+        mklabel gpt \
+        mkpart ESP fat32 8MiB 100MiB \
+        set 1 boot on \
+        mkpart BOOT fat32 100MiB ${bootSize} \
+        set 2 bls_boot on \
+        mkpart ROOT ext4 ${bootSize} -1
+      ${optionalString deterministic ''
+          sgdisk \
+          --disk-guid=97FD5997-D90B-4AA3-8D16-C1723AEA73C \
+          --partition-guid=1:C12A7328-F81F-11D2-BA4B-00A0C93EC93B  \
+          --partition-guid=2:BC13C2FF-59E6-4262-A352-B275FD6F7172  \
```

Comment:
This doesn't seem right. These are [Discoverable Partitions Specification](https://uapi-group.org/specifications/specs/discoverable_partitions_specification/) GUIDs, but you've used them for the partition GUIDs instead of the partition *type* GUIDs.

Reply:
Right makes sense now. I reused ids from the next type.